### PR TITLE
[IIIF-869] Move V2 manifest work into its own verticle

### PR DESCRIPTION
**Pull request by ksclarke**
* Add CODEOWNERS file
* Update dependencies and parent project
* Make CsvHeaders (de)serializable
* Remove the jiiify-presentation v2 classes and related logic from ManifestVerticle
* Create a V2ManifestVerticle and move v2 classes and the logic specific to them to it
* Move ManifestLabelComparator to V2ManifestLabelComparator
* Create stub V3ManifestVerticle and V3ManifestLabelComparator
* Update a couple of deprecated Future(s) to Promise(s)
* Update a few descriptive/stylistic things (variable names, etc.)

### CODE REVIEW

**Comment by ksclarke** [comment]
Seems the Codacy integration is still not working correctly. It's not using our Checkstyle and PMD config files. I've made the check "not required" temporarily and created a ticket to fix the problem: https://jira.library.ucla.edu/browse/IIIF-926

File: src/main/java/edu/ucla/library/iiif/fester/verticles/V2ManifestVerticle.java [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+        sendMessage(S3BucketVerticle.class.getName(), message, options, update -> {
+            if (update.succeeded()) {
+                aMessage.reply(collection.toJSON());
+            } else {
+                error(aMessage, update.cause(), MessageCodes.MFS_152, update.cause().getMessage());
+            }
+        });
+    }
+
+    private void updatePages(final Message<JsonObject> aMessage) throws JsonProcessingException {
```
Comment (markmatney) [inline src/main/java/edu/ucla/library/iiif/fester/verticles/V2ManifestVerticle.java:294]: missing Javadoc

File: src/main/java/edu/ucla/library/iiif/fester/verticles/V2ManifestVerticle.java [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+        });
+
+        aPromise.complete();
+    }
+
+    /**
+     * Create a new collection.
+     *
+     * @param aMessage An event queue message
+     * @throws JsonProcessingException If
```
Comment (markmatney) [inline src/main/java/edu/ucla/library/iiif/fester/verticles/V2ManifestVerticle.java:117]: if...

File: src/main/java/edu/ucla/library/iiif/fester/verticles/V2ManifestVerticle.java [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+            metadata.add(MetadataLabels.REPOSITORY_NAME, repositoryName);
+        });
+
+        getMetadata(workRow, csvHeaders.getLocalRightsStatementIndex()).ifPresent(localRightsStatement -> {
+            manifest.setAttribution(new Attribution(localRightsStatement));
+        });
+
+        getMetadata(workRow, csvHeaders.getRightsContactIndex()).ifPresent(rightsContract -> {
+            metadata.add(MetadataLabels.RIGHTS_CONTACT, rightsContract);
+        });
```
Comment (markmatney) [inline src/main/java/edu/ucla/library/iiif/fester/verticles/V2ManifestVerticle.java:211]: Nice! Much shorter and more readable than the old code

**Comment by markmatney** [comment]
also, the Codacy check is failing

**Comment by markmatney** [comment]
oh wait, you mentioned why already

**Pull request merged**